MBFaker/MBFakerInternet.m

```diff
@@ -28,20 +28,64 @@ + (NSString*)userName {
     
     for (int i=0; i<1+arc4random()%2;i++) {
         if (i == 0)
-            [components addObject:[MBFakerName firstName]];
+            [components addObject:[self fixUmlauts:[MBFakerName firstName]]];
         else
-            [components addObject:[MBFakerName lastName]];
+            [components addObject:[self fixUmlauts:[MBFakerName lastName]]];
     }
     
     return [[components componentsJoinedByString:@"."] lowercaseString];
 }
 
 + (NSString*)domainName {
-    return @"";
+    NSString *name = [NSString stringWithFormat: @"%@.%@", [self domainWord], [self domainSuffix]];
+    return name;
 }
 
 + (NSString*)domainWord {
-    return @"";
+    NSString *name = [(NSString*)[[MBFakerCompany name] componentsSeparatedByString: @" "][0] lowercaseString];
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"\\W" options:NSRegularExpressionCaseInsensitive error: NULL];
+    NSString *modifiedString = [regex stringByReplacingMatchesInString:name options:0 range:NSMakeRange(0, [name length]) withTemplate:@""];
+    
+    return [self fixUmlauts:modifiedString];
+}
```

Comment:
Also, in the original ruby version in the comments, they don't fix the umlauts here, but only in `domainName`.  However, `domainWord` is also part of the public API and it seems like it could have other uses, so I thought it made sense to do it here instead.